metabase/metabase
# scatter fix log x-axis bug

Closes https://github.com/metabase/metabase/issues/36588

### Description

This bug was caused by the `xAxis.boundaryGap` echarts option. For log and power x-axis scales in the scatter plot, this option should be `undefined`, as it is for the linear x-axis scale. This is because values can be touching the chart boundary, for example on the log scale, an x value of 1 is plotted directly on the y-axis in the log scale. When this happens, if there is a `boundaryGap` value, echarts will mess up the chart.

### How to verify

1. Create a scatter plot viz with any data point that has an x-value of 1
2. Change x-axis scale to log
3. Add the question to a dashboard, send a subscription
4. Confirm the chart renders correctly

### Demo

| Before  | After |
| ------------- | ------------- |
| <img width="579" alt="Screenshot 2023-12-17 at 10 54 36 AM" src="https://github.com/metabase/metabase/assets/37751258/37c7fd5e-4301-4cb5-8ad4-591af67ae696"> | <img width="585" alt="Screenshot 2023-12-17 at 10 51 04 AM" src="https://github.com/metabase/metabase/assets/37751258/a85a900a-0af2-491d-b0cc-d5a83d1604c6"> |

### Checklist
